lib/OpenQA/WebAPI/Plugin/REST.pm

```diff
@@ -2,49 +2,33 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 
 package OpenQA::WebAPI::Plugin::REST;
-use Mojo::Base 'Mojolicious::Plugin';
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
 
-use Scalar::Util ();
-use Carp ();
-
-sub register {
-
-    my ($self, $app, $config) = @_;
+use Scalar::Util 'blessed';
+use Carp 'croak';
 
+sub register ($self, $app, $config) {
     # special anchor tag with data-method
     $app->helper(
-        action_link => sub {
-            my ($self, $method, $content) = (shift, shift, shift);
+        action_link => sub ($self, $method, $content, @args) {
             my $url = $content;
-
-            if ($self->is_operator) {
-                # Content
-                unless (ref $_[-1] eq 'CODE') {
-                    $url = shift;
-                    push @_, $content;
-                }
-
-                Carp::croak "url is not a url"
-                  unless Scalar::Util::blessed $url && $url->isa('Mojo::URL');
-
-                return $self->tag('a', href => $url, 'data-method' => $method, @_);
-            }
-            else {
-                return '';
+            return '' unless $self->is_operator;
+            # Content
+            unless (ref $_[-1] eq 'CODE') {
+                $url = shift @args;
+                push @args, $content;
             }
+            croak "url is not a url" unless blessed $url && $url->isa('Mojo::URL');
+
+            return $self->tag('a', href => $url, 'data-method' => $method, @args);
         });
 
     # special anchor tag for post links
-    $app->helper(
-        link_post => sub {
-            my $self = shift;
-            $self->action_link('post', @_);
-        });
+    $app->helper(link_post => sub ($self, @args) { $self->action_link('post', @args) });
```

Comment:
And `$self` instead of `$c` again.

Reply:
thx, see https://github.com/os-autoinst/openQA/pull/4419